src/extractor/guidance/mergable_road_detector.cpp

```diff
@@ -0,0 +1,427 @@
+                                  selector);
+    }
+
+    BOOST_ASSERT(!left_accumulator.intersection.empty() && !right_accumulator.intersection.empty());
+
+    // find the closes resembling a right turn
+    const auto connector_turn = left_accumulator.intersection.findClosestTurn(90);
+    /* check if that right turn connects to the right_accumulator intersection (i.e. we have a
+     * triangle)
+     * a connection should be somewhat to the right, when looking at the left side of the
+     * triangle
+     *
+     *    b ..... c
+     *     \     /
+     *      \   /
+     *       \ /
+     *        a
+     *
+     * e.g. here when looking at `a,b`, a narrow triangle should offer a turn to the right, when
+     * we want to connect to c
+     */
+    if (angularDeviation(connector_turn->angle, 90) > NARROW_TURN_ANGLE)
+        return false;
+
+    const auto num_lanes = [this](const MergableRoadData &road) {
+        return std::max<std::uint8_t>(
+            node_based_graph.GetEdgeData(road.eid).road_classification.GetNumberOfLanes(), 1);
+    };
+
+    // the width we can bridge at the intersection
+    const auto assumed_road_width = (num_lanes(lhs) + num_lanes(rhs)) * ASSUMED_LANE_WIDTH;
+    const auto distance_between_triangle_corners = util::coordinate_calculation::haversineDistance(
+        node_coordinates[node_based_graph.GetTarget(left_accumulator.via_edge_id)],
+        node_coordinates[node_based_graph.GetTarget(right_accumulator.via_edge_id)]);
+    if (distance_between_triangle_corners > (assumed_road_width + 10))
+        return false;
+
+    // check if both intersections are connected
+    IntersectionFinderAccumulator connect_accumulator(5, intersection_generator);
+    graph_walker.TraverseRoad(node_based_graph.GetTarget(left_accumulator.via_edge_id),
+                              connector_turn->eid,
+                              connect_accumulator,
+                              selector);
+    // the if both items are connected
+    return node_based_graph.GetTarget(connect_accumulator.via_edge_id) ==
+           node_based_graph.GetTarget(right_accumulator.via_edge_id);
+}
+
+bool MergableRoadDetector::HaveSameDirection(const NodeID intersection_node,
+                                             const MergableRoadData &lhs,
+                                             const MergableRoadData &rhs) const
+{
+    if (angularDeviation(lhs.bearing, rhs.bearing) > 95)
+        return false;
+
+    // Find a coordinate following a road that is far away
+    NodeBasedGraphWalker graph_walker(node_based_graph, intersection_generator);
+    const auto getCoordinatesAlongWay = [&](const EdgeID edge_id, const double max_length) {
+        LengthLimitedCoordinateAccumulator accumulator(
+            coordinate_extractor, node_based_graph, max_length);
+        SelectStraightmostRoadByNameAndOnlyChoice selector(
+            node_based_graph.GetEdgeData(edge_id).name_id, lhs.bearing, false);
```

Comment:
Hm here and bove I had to check the selector what the boolean false is all about .. what do you think about making as much enums out of bools as possible or at least on the callsite doing something like `/*pred*/=false`?